src/Instance.cs

```diff
@@ -1,136 +1,214 @@
 using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Dynamic;
-using Wasmtime.Externs;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
 
 namespace Wasmtime
 {
     /// <summary>
     /// Represents an instantiated WebAssembly module.
     /// </summary>
-    public class Instance : DynamicObject, IDisposable, IImportable
+    public class Instance : IExternal
     {
         /// <summary>
-        /// The exported functions of the instance.
+        /// Creates a new WebAssembly instance.
         /// </summary>
-        public IReadOnlyList<ExternFunction> Functions => _externs.Functions;
+        /// <param name="context">The store context to create the instance in.</param>
+        /// <param name="module">The module to create the instance for.</param>
+        /// <param name="imports">The imports for the instance.</param>
+        public Instance(StoreContext context, Module module, params object[] imports)
+        {
+            if (module is null)
+            {
+                throw new ArgumentNullException(nameof(module));
+            }
 
-        /// <summary>
-        /// The exported globals of the instance.
-        /// </summary>
-        public IReadOnlyList<ExternGlobal> Globals => _externs.Globals;
+            if (imports is null)
+            {
+                throw new ArgumentNullException(nameof(imports));
+            }
 
-        /// <summary>
-        /// The exported tables of the instance.
-        /// </summary>
-        public IReadOnlyList<ExternTable> Tables => _externs.Tables;
+            unsafe
+            {
+                var externs = stackalloc Extern[imports.Length];
+                for (int i = 0; i < imports.Length; ++i)
+                {
+                    var external = imports[i] as IExternal;
+                    if (external is null)
+                    {
+                        throw new ArgumentException($"Objects of type `{imports[i].GetType().ToString()}` cannot be imported.");
+                    }
+                    externs[i] = external.AsExtern();
+                }
+
+                var error = Native.wasmtime_instance_new(context.handle, module.NativeHandle, externs, (UIntPtr)imports.Length, out this.instance, out var trap);
```

Comment:
I had more of a comment in `Linker` below, but this is another point where wasm code is executed

Reply:
Opened issue #63 for this.